docs/blog/posts/constitutional-ai-or-validation.md

```diff
@@ -0,0 +1,221 @@
+---
+draft: False 
+date: 2023-10-17
+tags:
+    - pydantic
+    - validation
+    - guardrails 
+    - constitutional ai
+---
+
+# Constitutional AI is just Probabilistic Validation
+
+Earlier this year, there was a lot of discussion about Constitutional AI and its role in guiding models. There was also talk about the need for guard rails and special systems to ensure safe outputs. However, I found myself unimpressed. Why create a new term when we already have one that describes the concept? That term is "validation," and now we can do it probabilistically.
```

Comment:
[NIT]

> That term is "validation," and now we can do it probabilistically.

Typically, we think of validation as being deterministic with a clear pass or fail. Is the date in the right format, or is the response within a certain length? In this case, I agree that validation and guardrails are synonymous, at least in the [original vision of the latter](https://github.com/guardrails-ai/guardrails#-rail-spec).

On the other hand, I think Constitutional AI, and your example in line 84, is subjective and thus probabilistic, as you rightly pointed out. Thus, I wonder if calling this validation would be overloading the term?

I like how you named it `llm_validator` as it mirrors `field_validator`. Nonetheless, I wonder if the following suggestions would help users better distinguish between validation as deterministic vs. moderation as subjective/probabilistic:
- `probabilistic_validator`: To make clear that it's not deterministic. Also, allows for non-LLM validations.
- `llm_moderator`: To make clear that it's not validation

Either way, not a one-way door. Even if `Instructor` sticks to `llm_validator`, it's a simple name change to update in future.

Reply:
Yeah even fuzzy validator might make more sense.